### ISSUE TITLE
[Draft] Typescript

### ISSUE THREAD

**Issue by sidferreira** [tedconf/js-crushinator-helpers#69]
This is a small draft of types for the `crushinator`. I quickly wrote so feel free to tag me to "finish and PR it" (just not today).

The `EnumLiteralsOf` should be a way to have both an enum and string working, but was failing this time and I don't have time to review ATM.

```
export type EnumLiteralsOf<T extends object> = T[keyof T];
export type ICrushinatorAlign = EnumLiteralsOf<typeof ICrushinatorAlign>;
export const ICrushinatorAlign = Object.freeze({
  TOP: 'top' as 'top',
  BOTTOM: 'bottom' as 'bottom',
  LEFT: 'left' as 'left',
  RIGHT: 'right' as 'right',
  MIDDLE: 'middle' as 'middle',
});

interface ICrushinatorCropConfig {
  width: number;
  height: number;
  x: number;
  y: number;
}

interface ICrushinatorConfig {
  width?: number; // Target image width in pixels.	width: 320
  height?: number; // Target image height in pixels.	height: 180
  quality?: number; // Image quality as a percentage (0-100). Defaults to 82.	quality: 74
  fit?: boolean; // True to zoom and crop the image for best fit into the target dimensions (width and height) if both are provided. Enabled by default.	fit: true
  defaults?: boolean; // False to disable the default options added by the helper.	defaults: false
  crop?: ICrushinatorCropConfig; // An image crop configuration specifying the dimensions and coordinates of the section to crop.	crop: { width: 200, height: 100, x: 50, y: 50 }
  align?: ICrushinatorAlign | string; // When cropping an image, it can be aligned to the "top", "bottom", "left", "right", or "middle" of the crop frame.	align: 'top'
  blur?: number; // A blur spread amount in pixels.	blur: 4
  gamma?: number; // A gamma correction multiplier. Values less than 1.0 darken the image, and values greater than 1.0 lighten it.	gamma: 1.4
  grayscale?: boolean; // Fully desaturates the image.	grayscale: true
  unsharp?: boolean; // Applies an unsharp mask to accentuate image details. Enabled by default.	unsharp: true
  query?: any; // Additional query parameters to include in the Crushinato
}
```

**Comment by redoPop** [comment]
I think @VinSpee beat you to the punch in #41 but we haven't cut a release since merging: d'oh! 🤦 Sorry about that.

**Comment by sidferreira** [comment]
@redoPop any chance of releasing it in the coming minutes? :joy:

**Comment by redoPop** [comment]
Absolutely! Just published v2.8.0.

I'm closing this issue for now but please do open a PR if you'd like to see changes to the current type definitions. Vince will a better reviewer than me for such a PR: I'm not using this library in any TypeScript projects yet, so I can't speak to any conflicts changes might create.

Thanks again!

**Comment by sidferreira** [comment]
@redoPop ( cc @VinSpee ) working perfectly
